Automattic/wp-calypso
# Jetpack Partner Portal: Introduce adding credit card as payment method

Design: JVpmiQH74FKEIrPLjqv8p7-fi-1262%3A13812


#### Changes proposed in this Pull Request

* This PR is a port of the composite checkout in the Calypso Green Partner Portal to add credit cards to stripe.
* A lot of this was copied from /me/purchases and adapted to work with Jetpack Cloud. We've trimmed most of what's not needed.
* The PR is functional and the design is almost the same as the proposed, but there are still a few things that can be improved

_Note: this patch currently uses the WPCOM Stripe account, and a later iteration is needed in order to add the separate context for the Payment Methods_

#### Testing instructions

- Check out this PR and run `yarn start-jetpack-cloud`.
- Go to http://jetpack.cloud.localhost:3000/partner-portal/payment-method/add
- Click to add a new card
- Make sure you can go through the process smoothly and that the card is added



![image](https://user-images.githubusercontent.com/5714212/125355316-40858580-e33b-11eb-982d-466f71f42862.png)
